### DISCUSSION TITLE
1.0.0 Release Plans

### DISCUSSION THREAD

**Discussion by ArchLeaders** [NX-Editor/.github#3] [Announcements]
Some time in the future I will actually release the official 1.0.0 version of NX-Editor. Before that happens though, there's a couple things that (should) to be done. I mostly have this here for my sake, but I'm open to contributions as well.

School will also be starting soon, so release may be a while still, but I'll be aiming for pre-2024.

## General

- Some form of documentation for getting started
- Better logging or errors and status changes
- Fix memory leaks caused by [AvaloniaEdit](https://github.com/AvaloniaUI/AvaloniaEdit/issues/362)

## Styling

- Vertical bar to separate standard menu items and plugin items?

## Plugins

- EPD Plugin
  - ~~Finish RESTBL/RSTB documentation~~
  
---

If you find any other bugs in the current version of NX-Editor, please leave an [issue](https://github.com/NX-Editor/NxEditor/issues/new) on the [GitHub repository](https://github.com/NX-Editor/NxEditor).

**Comment by Spletz47** [comment]
Is this project meant to be almost a continuation of Switch Toolbox after its development ended recently?

**Reply by ArchLeaders** [reply]
Not really, this was in development long before Toolbox reached EoL. It may end up being the next best thing, but I doubt it will reach the same support that Toolbox has unless I find a lot more time.

**Reply by Spletz47** [reply]
Oh yeah, right, sorry. I looked again and I see the initial release date, lol. Well, there are many modern Switch files that I would've hoped to be readable (including byaml v7 which you've already taken care of) by Toolbox. It would be great to see files like belnk.zs, bslnk.zs, and esetb.byml.zs (that I've seen in Splatoon 3) be supported at some point since I'd kill to take a look through those. I swear NX Editor actually let me open the last one I listed and I was looking through it thinking this is great, only for it to crash and it will not let me open it again, saying it's not supported. Weird, but whatever. Anyway, sorry for yapping. Just some thoughts I had. Thanks for this tool!

**Reply by ArchLeaders** [reply]
I'd like to add XLink and PCTL support if I get more time, unfortunately, that's hard to come by.

Also `esetb.byml.zs` is supported, it's just a BYML v7 file with an embedded PTCL. I'm not sure why it would crash though.

**Reply by Spletz47** [reply]
Yeah, that's what I figured. But this is the log when I try to open it now

Loaded NxEditor.EpdPlugin.dll
14:51:54:165 [LoadPlugin]: Loaded NxEditor.EpdPlugin.dll
Loaded NxEditor.TotkPlugin.dll
14:51:54:180 [LoadPlugin]: Loaded NxEditor.TotkPlugin.dll
[Info] Dictionaries not found, skipping...
[Info] Dictionaries not found, skipping...
[Info] Dictionaries not found, skipping...
14:52:26:561 [LoadEditor]: Processing Sdodr.Product.700.Nin_NX_NVN.esetb.byml.zs
14:52:26:625 [TryLoadEditor]: |-
  System.NotSupportedException: The provided IFileHandle is not a supported data type
     at NxEditor.PluginBase.ServiceLoader.RequestService(IEditorFile handle)
     at NxEditor.Components.EditorManager.LoadEditor(IEditorFile handle) in D:/a/NxEditor/NxEditor/src/NxEditor/Components/EditorManager.cs:line 38
     at NxEditor.Components.EditorManager.TryLoadEditor(IEditorFile handle) in D:/a/NxEditor/NxEditor/src/NxEditor/Components/EditorManager.cs:line 20

**Reply by ArchLeaders** [reply]
Make sure your game path is set. It looks like it can't find the dictionaries in the current configuration.

> `[Info] Dictionaries not found, skipping...`
> `[Info] Dictionaries not found, skipping...`
> `[Info] Dictionaries not found, skipping...`

**Reply by Spletz47** [reply]
Ah, I feel quite stupid now. Thank you very much. It now I discover a list of effects but then a massive string of letters at the end. That's probably why it originally crashed too lol. If you do decide to add support for those other file formats, just letting you know that I am more than willing to donate to this project since it is the best hope we have at reading these files right now. Thanks again!

**Reply by Vulunta** [reply]
The "letters" are a bas64 encoded PTCL file. You can convert them to binary online, copy the data into a hex editor and then open the files in Toolbox. The file extension may need to be .ptcl as well, not 100% sure though.

**Reply by ArchLeaders** [reply]
> Ah, I feel quite stupid now. Thank you very much. It now I discover a list of effects but then a massive string of letters at the end. That's probably why it originally crashed too lol.

I'm not sure why that would cause a crash. It could be a memory overload, although I've never had the issue before.

> If you do decide to add support for those other file formats, just letting you know that I am more than willing to donate to this project since it is the best hope we have at reading these files right now.

I might add donations when I turn 18, I don't need the money though so I might not bother.

Thanks for the encouragement though, hopefully I can find time to work on more file formats.
